### PR TITLE
Migrate Admin pages to React

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature

## Description

- [x] Migrate System Status page
- [x] Refine Celery Status page
- [x] Migrate Outdated Queries page

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

image

image

image

### CODE REVIEW

**Comment by ranbena** [comment]
Wow 1st look, looks so good! 👏
Great use of flexbox.

Few things I noticed:
1. System Status > Queues > Celery doesn't display a tooltip on hover.
2. Outdated Queries > "Last updated" value is "few seconds" on master, and "49 years ago" in this.
3. Outdated Queries - when there are no queries, no need to show the sidebar and the "Sorry, we couldn't find anything." isn't a good fit for this situation.

**Comment by kravets-levko** [comment]
@ranbena Thanks! All fixed.

File: client/app/components/admin/CeleryStatus.jsx [hunk 1]
```diff
@@ -46,7 +46,14 @@ function parseTasks(tasks) {
 function QueuesTable({ loading, queues }) {
   const columns = ['Name', 'Active', 'Reserved', 'Waiting'].map(c => ({ title: c, dataIndex: c.toLowerCase() }));
 
-  return <Table columns={columns} rowKey="name" dataSource={queues} loading={loading} />;
+  return (
+    <Table
+      columns={columns}
+      rowKey="name"
+      dataSource={queues}
+      loading={{ spinning: loading, indicator: <i className="fa fa-spinner fa-pulse" /> }}
```
Comment (arikfr) [inline client/app/components/admin/CeleryStatus.jsx]: This is definitely better than Ant's default (ugly) indicator :-)

Can we use [`Spin.setDefaultIndicator`](https://ant.design/components/spin/#Static-Method) to globally set this?

File: client/app/components/admin/Layout.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import cx from 'classnames';
+import { PageHeader } from '@/components/PageHeader';
+
+export default function Layout({ activeTab, children }) {
+  return (
+    <div className="container">
+      <PageHeader title="Admin" />
+
+      <div className="bg-white tiled">
+        <ul className="tab-nav">
```
Comment (arikfr) [inline client/app/components/admin/Layout.jsx]: Can't we use Ant's tabs here?

File: client/app/components/admin/StatusBlock.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          itemLayout="vertical"
+          dataSource={info}
+          renderItem={([name, queue]) => (
+            <List.Item extra={<span className="badge">{queue.size}</span>}>
+              {name}
+              <Tooltip
+                className="m-l-5"
+                title={isString(queue.data_sources) && (queue.data_sources !== '') ? queue.data_sources : 'n/a'}
+                mouseLeaveDelay={0}
+              >
```
Comment (arikfr) [inline client/app/components/admin/StatusBlock.jsx]: The Tooltip is not needed -- we no longer send the data sources list.

**Comment by arikfr** [comment]
image

DB size was humanized before.

File: client/app/components/admin/StatusBlock.jsx (hunk 3 above)
Comment (kravets-levko) [reply client/app/components/admin/StatusBlock.jsx]: On preview instance it works :thinking_face:

File: client/app/components/admin/Layout.jsx (hunk 2 above)
Comment (kravets-levko) [reply client/app/components/admin/Layout.jsx]: If tab header can be a link and have an empty body - yes. I'll check

File: client/app/components/admin/StatusBlock.jsx (hunk 3 above)
Comment (arikfr) [reply client/app/components/admin/StatusBlock.jsx]: Because it doesn't run the latest version of the code :) Try locally.

**Comment by ranbena** [comment]
Suggestion for future - make these into Redash built-in dashboards (grids, counts). 🤘

File: client/app/components/admin/Layout.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import PropTypes from 'prop-types';
+import Tabs from 'antd/lib/tabs';
+import { PageHeader } from '@/components/PageHeader';
+
+import './layout.less';
+
+export default function Layout({ activeTab, children }) {
+  return (
+    <div className="container admin-page-layout">
+      <PageHeader title="Admin" />
```
Comment (ranbena) [inline client/app/components/admin/Layout.jsx]: ux nit: Shouldn't this be titled "System Status"?

**Comment by kravets-levko** [comment]
@ranbena That idea crossed my mind some time ago :slightly_smiling_face: We'll need some new visualization types for that.

File: client/app/components/admin/CeleryStatus.jsx [hunk 5]
```diff
@@ -46,7 +46,14 @@ function parseTasks(tasks) {
 function QueuesTable({ loading, queues }) {
```
Comment (ranbena) [inline client/app/components/admin/CeleryStatus.jsx]: Perhaps an opportunity to fix the "forEach" error thrown when there're no celery tasks.
```js
function parseTasks(tasks = []) {
```

**Comment by ranbena** [comment]
> @ranbena That idea crossed my mind some time ago 🙂 We'll need some new visualization types for that.

It could be a great forcing function to make better widgets and set an example to our clients.

**Comment by arikfr** [comment]
> It could be a great forcing function to make better widgets and set an example to our clients.

I agree. Let's start moving in this direction when our visualizations are pure React, so it's easier to create better APIs for them. :-)

File: client/app/components/admin/Layout.jsx (hunk 4 above)
Comment (kravets-levko) [reply client/app/components/admin/Layout.jsx]: No, "System Status" is the first tab

**Comment by kravets-levko** [comment]
@ranbena (cc @arikfr) I fixed bug with `forEach` and refactored that page a bit more: sometimes it tried to call `setState` on unmounted component (I fixed that as well) and refined code - created more smaller components, and moved main logic to main (page) component.

**Comment by kravets-levko** [comment]
@ranbena Which tabs do you mean? There are top-level tabs ("System Status", "Celery Status", ...) and inner tabs on "Celery Status". For top-level tabs I had to disable animation because that tabs are link, and when clicked - animation started, then component destroyed - so it looked really weird. For inner tabs animation doesn't look so weird.

**Comment by ranbena** [comment]
> @ranbena Which tabs do you mean? 

Celery tabs

> For inner tabs animation doesn't look so weird.

Okey dokey

**Comment by kravets-levko** [comment]
Decided to disable animation - for consistency between top-level and inner tabs.

**Comment by kravets-levko** [comment]
@arikfr @gabrieldutra If you're fine with this - LMK if I can merge it. Thanks!

**Pull request merged**